src/java.desktop/macosx/native/libawt_lwawt/awt/CSystemColors.m

```diff
@@ -126,7 +126,7 @@ + (void)reloadColors {
 + (NSColor*)getColor:(NSUInteger)colorIndex useAppleColor:(BOOL)useAppleColor {
     NSColor* result = nil;
 
-    if (colorIndex < (useAppleColor) ? sun_lwawt_macosx_LWCToolkit_NUM_APPLE_COLORS : java_awt_SystemColor_NUM_COLORS) {
+    if (colorIndex < ((useAppleColor) ? sun_lwawt_macosx_LWCToolkit_NUM_APPLE_COLORS : java_awt_SystemColor_NUM_COLORS)) {
```

Comment:
This looks like a plain old bug fix, nothing really to do with the compiler upgrade.  The new compiler presumably just has a new warning that brought attention to the problem.  As such, I don't think it belongs in a PR that's about issues related to the compiler upgrade.  Someone might want to backport just this fix, for example.

Reply:
Hello  Kim, Paul -  so should we open a separate bug for the src/java.desktop/macosx/native/libawt_lwawt/awt/CSystemColors.m issue because it might be a general  problem just uncovered by the new compiler ?
Paul , do you want to do this ? Or should I open a bug and post a separate change for the useAppleColor check in CSystemColors.m ?

Reply:
I created 
https://bugs.openjdk.java.net/browse/JDK-8253791
8253791: Issue with useAppleColor check in CSystemColors.m
for this issue (Kim and Paul are fine to have a separate JBS issue for this)